[Yik Jun Xian] iP

yikjunxian opened the pull request:
## Gumbo
> If you set your goals ridiculously high and it's a failure, you will fail above everyone else's success. - James Cameron ([source](https://blog.hubspot.com/sales/famous-quotes)).

Don't be a Dumbo and use Gumbo! 
Gumbo frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ _SUPER_ FAST to use

All you need to do is,

1. download it from [here](https://github.com/yikjunxian/ip).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉 
And it is ***FREE***!

Features:

- [X] Managing tasks
- [x] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)


If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Commit:
Rename main class

Commit:
Add gretting

Commit:
Add support for echo

Commit:
Add support for list

Commit:
Add support for mark as done and undone

Commit:
Added specific tasks Todo, Event and Deadline

Commit:
Added automated text UI testing

Commit:
Added automated test UI Testing

Commit:
Added incorrect input error handling

Commit:
Added support for deleting task

Commit:
Reorganise local classes into separate files

Commit:
Add Deadline class

Commit:
Add Event class

Commit:
Add Task class

Commit:
Add Todo class

Commit:
Add GumboException class

Commit:
Add support to load and save task on hard disk

Commit:
Add method to convert task to specific format required in .txt file

Commit:
Add implementation of toTextString method

Commit:
Add implementation of toTextString method

Commit:
Add implementation of toTextString method

Commit:
Add support for dates and times in tasks

Commit:
Add support for dates and times in Deadline

Commit:
Add support for dates and times in Event

Commit:
Merge branch 'branch-Level-8'

Commit:
Simplify main logic of Gumbo using Ui, Storage, Parser and Command objects

Commit:
Add TaskList object that contains the task list and support operations on it

Commit:
Add Parser object that deals with making sense of user commands

Commit:
Add Storage object that supports loading of tasks from file and saving tasks in file

Commit:
Add Ui object that deals with interaction with the user

Commit:
Add GumboException that is the super class of all custom exceptions in the chatbot

Commit:
Add IllegalValueException that extends GumboException

Commit:
Add AddCommand class that extends Command

Commit:
Add Command class that is the super class of all types of user commands

Commit:
Add DeleteCommand class that extends Command

Commit:
Add MarkCommand class that extends Command

Commit:
Add UnmarkCommand class that extends Command

Commit:
Add ListCommand class that extends Command

Commit:
Add ExitCommand class that extends Command

Commit:
Move Task class in task package

Commit:
Move Deadline class in task package

Commit:
Move Event class in task package

Commit:
Move Todo class in task package

Commit:
Merge branch 'branch-MoreOOP'

Commit:
Rename Gumbo class to Main class that executes the main logic of the chatbot

Commit:
Add Gumbo file which contains all the Gumbo packages

Commit:
Moved Gumbo packages under Gumbo file

Commit:
Merge branch 'branch-packages'

Commit:
Merge branch 'master' into add-gradle-support

Commit:
Add Gumbo package and moved all sub-packages under it

Commit:
Add checkstyle to build.gradle file

Commit:
Add gradlew

Commit:
Add graddle wrapper

Commit:
Add checkstyle config files

Commit:
Merge branch 'add-gradle-support'

Commit:
Remove redundant import statement

Commit:
Add JUnit tests for methods in AddCommand and Storage

Commit:
Merge branch 'JUnit'

Commit:
Add JavaDoc comments to non-trivial classes and methods

Commit:
Tweak code to comply with coding standard

Commit:
Tweak code to comply with coding standards

Commit:
Add support for find command

Commit:
revert commit committed to wrong branch

Commit:
Revert last wrong commit

kellywsq03 commented on src/main/java/gumbo/commands/DeleteCommand.java:
```diff
@@ -0,0 +1,39 @@
+    public void execute(Ui ui, Storage storage, TaskList taskList) {
+        Task deletedTask = taskList.get(taskNumToDelete);
+        taskList.remove(taskNumToDelete);
+        ui.showToUser("Noted. I've removed this task:\n"
```
Perhaps you may want to abstract the message segment and place it in the UI class for better OOP

kellywsq03 commented on src/main/java/gumbo/parser/Parser.java:
```diff
@@ -0,0 +1,124 @@
+     * @param fullCommand The full string of user input.
+     * @return A {@code Command} object representing the user's request.
+     */
+    public static Command parse(String fullCommand) {
```
Consider using switch-case statements!

kellywsq03 commented on src/main/java/gumbo/storage/Storage.java:
```diff
@@ -0,0 +1,170 @@
+     *
+     * @throws InvalidStorageFilePathException if the default file path is invalid.
+     */
+    public Storage() throws InvalidStorageFilePathException {
```
Good use of error handling

kellywsq03 reviewed (commented):
LGTM! Good use of JavaDoc headers and suitable naming conventions.

ryenl commented on src/main/java/gumbo/parser/Parser.java:
```diff
@@ -0,0 +1,124 @@
+     * @return A {@code Command} object representing the user's request.
+     */
+    public static Command parse(String fullCommand) {
+        if (fullCommand.equals("bye")) {
```
The current implementation uses a series of if-else statements. Would it be cleaner to switch to a switch-case statement improve readability and organization?

ryenl commented on src/test/java/gumbo/storage/StorageTest.java:
```diff
@@ -0,0 +1,66 @@
+import java.nio.file.Paths;
+import java.util.ArrayList;
+
+import static org.junit.jupiter.api.Assertions.*;
```
Imported classes should always be listed explicitly.

ryenl commented on src/main/java/gumbo/tasks/Event.java:
```diff
@@ -0,0 +1,37 @@
+
+public class Event extends Task {
+
+    protected LocalDate to;
+    protected LocalDate from;
+
+    private DateTimeFormatter inputDateFormat = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+    private DateTimeFormatter outputDateFormat = DateTimeFormatter.ofPattern("MMM dd yyyy");
```
good use of separation of logical units!

ryenl commented on src/main/java/gumbo/tasks/Event.java:
```diff
@@ -0,0 +1,37 @@
+            this.to = LocalDate.parse(to, inputDateFormat);
+            this.from = LocalDate.parse(from, inputDateFormat);
+        } catch (DateTimeParseException e) {
+            System.out.println("Invalid date format. Please input date format as [yyyy-MM-dd].");
```
could implement as a method of the Ui class to show this error?

ryenl reviewed (commented):
LGTM! Just some minor nits 💯

Commit:
Tweak file path argument when creating Main class

Commit:
Add GUI to application

Commit:
Tweak code to adhere to checkstyle

Commit:
Merge branch 'branch-Level-10'

Commit:
Add assertion statements

Commit:
Remove redundant run() method

Commit:
Fix bug on assertion statement in execute() method

Commit:
String::parse method is too long.

A lengthy parse method with plenty of else-if statements
makes it tedious and time consuming for readers to understand
the types of inputs being parsed.

Let's shorten the method by using switch-case statements for
the different types of inputs.

This allows the method to be cleaner and more understandable.

Commit:
Implement getEvent method when parsing user input following SLAP principle

yikjunxian committed:
Merge pull request #1 from yikjunxian/branch-A-Assertions

Branch-A-Assertions

yikjunxian committed:
Merge pull request #2 from yikjunxian/branch-A-CodeQuality

Branch-A-CodeQuality

Commit:
Update command does not exist.

Having an update command enables users to update specific
details of their task without having to delete and add
their task again which is troublesome.

Lets add a new UpdateCommand that inherits Command.
When executed, this command will update the relevant
task in the task list.

This allows users to update existing tasks by
specifying which task to update and the detail to
update, such as the description or date.

Commit:
Merge branch 'branch-C-Update'

Commit:
Fix bug in exception handling in Event and Deadline classes

Commit:
Remove text-based UI

Commit:
Tweaked UI design to make chatbot window larger

Commit:
Add screenshot of GUI window

Commit:
Add README file

Commit:
Tweak output message for find command

Commit:
Fix bug in Junit tests

Commit:
Add some JavaDoc
